core/scripts/chaincli/handler/keeper_verifiable_load.go

```diff
@@ -0,0 +1,232 @@
+		From:        k.fromAddr,
+		Context:     ctx,
+		BlockNumber: big.NewInt(int64(blockNum)),
+	}
+
+	// get all active upkeep IDs on this verifiable load contract
+	upkeepIds, err := v.GetActiveUpkeepIDs(opts, big.NewInt(0), big.NewInt(0))
+	if err != nil {
+		log.Fatalf("failed to get active upkeep IDs from %s: %v", k.cfg.VerifiableLoadContractAddress, err)
+	}
+
+	upkeepStats := &UpkeepStats{BlockNumber: blockNum}
+
+	resultsChan := make(chan *UpkeepInfo, MaxUpkeepNum)
+	idChan := make(chan *big.Int, MaxUpkeepNum)
+
+	var wg sync.WaitGroup
+
+	// create a number of workers to process the upkeep ids in batch
+	for i := 0; i < WorkerNum; i++ {
+		wg.Add(1)
+		go k.getUpkeepInfo(idChan, resultsChan, v, opts, &wg)
+	}
+
+	for _, id := range upkeepIds {
+		idChan <- id
+	}
+
+	close(idChan)
+	wg.Wait()
+
+	close(resultsChan)
+
+	for info := range resultsChan {
+		upkeepStats.AllInfos = append(upkeepStats.AllInfos, info)
+		upkeepStats.TotalPerforms += info.TotalPerforms
+		upkeepStats.TotalDelayBlock += info.TotalDelayBlock
+		upkeepStats.SortedAllDelays = append(upkeepStats.SortedAllDelays, info.SortedAllDelays...)
+	}
+
+	sort.Float64s(upkeepStats.SortedAllDelays)
+
+	log.Println("\n\n================================== ALL UPKEEPS SUMMARY =======================================================")
+	p50, _ := stats.Percentile(upkeepStats.SortedAllDelays, 50)
+	p90, _ := stats.Percentile(upkeepStats.SortedAllDelays, 90)
+	p95, _ := stats.Percentile(upkeepStats.SortedAllDelays, 95)
+	p99, _ := stats.Percentile(upkeepStats.SortedAllDelays, 99)
+	maxDelay := upkeepStats.SortedAllDelays[len(upkeepStats.SortedAllDelays)-1]
+	log.Printf("For total %d upkeeps: total performs: %d, p50: %f, p90: %f, p95: %f, p99: %f, max delay: %f, total delay blocks: %f, average perform delay: %f\n", len(upkeepIds), upkeepStats.TotalPerforms, p50, p90, p95, p99, maxDelay, upkeepStats.TotalDelayBlock, upkeepStats.TotalDelayBlock/float64(upkeepStats.TotalPerforms))
+	log.Printf("All STATS ABOVE ARE CALCULATED AT BLOCK %d", blockNum)
+}
+
+func (k *Keeper) getUpkeepInfo(idChan chan *big.Int, resultsChan chan *UpkeepInfo, v *verifiable_load_upkeep_wrapper.VerifiableLoadUpkeep, opts *bind.CallOpts, wg *sync.WaitGroup) {
+	defer wg.Done()
+
+	for id := range idChan {
+		// fetch how many times this upkeep has been executed
+		c, err := v.Counters(opts, id)
+		if err != nil {
+			log.Fatalf("failed to get counter for %s: %v", id.String(), err)
+		}
+
+		// get all the buckets of an upkeep. 100 performs is a bucket.
+		b, err := v.Buckets(opts, id)
+		if err != nil {
+			log.Fatalf("failed to get current bucket count for %s: %v", id.String(), err)
+		}
+
+		info := &UpkeepInfo{
+			ID:                    id,
+			Bucket:                b,
+			TotalPerforms:         c.Uint64(),
+			DelayBuckets:          map[uint16][]float64{},
+			DelayTimestampBuckets: map[uint16][]float64{},
+		}
+
+		var delays []float64
+		var wg1 sync.WaitGroup
+		for i := uint16(0); i <= b; i++ {
+			wg.Add(1)
```

Comment:
Oh this is wrong; I think it needs to be `wg1.Add()`?

Reply:
done